### PR TITLE
CVSL-1216 updating dependencies

### PR DESCRIPTION
**Pull request by andrewrlee**
Including:
* Moving to java 19
* Bumping gradle
* Moving to kotlin dsl for specifying web security

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/integration/AuditIntegrationTest.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ class AuditIntegrationTest : IntegrationTestBase() {
       .expectBody(ErrorResponse::class.java)
       .returnResult().responseBody
 
-    assertThat(result?.userMessage).contains("Access is denied")
+    assertThat(result?.userMessage).contains("Access Denied")
```
Comment (andrewrlee) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/integration/AuditIntegrationTest.kt:36]: This is the spring defined message and it's changed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/config/ResourceServerConfiguration.kt [hunk 2]
```diff
@@ -5,40 +5,47 @@ import org.springframework.cache.CacheManager
...
+import org.springframework.security.config.annotation.web.invoke
 import org.springframework.security.config.http.SessionCreationPolicy
 import org.springframework.security.oauth2.jwt.JwtDecoder
 import org.springframework.security.oauth2.jwt.NimbusJwtDecoder
 import org.springframework.security.web.SecurityFilterChain
 
 @Configuration
 @EnableWebSecurity
-@EnableGlobalMethodSecurity(prePostEnabled = true, proxyTargetClass = true)
+@EnableMethodSecurity(prePostEnabled = true)
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/config/ResourceServerConfiguration.kt:19]: All the changes in this file represent moving away from deprecated code

**Review by nishanthmahasamudram — APPROVED** [review]
LGTM

**Review by nishanthmahasamudram — APPROVED** [review]
LGTM

**Pull request merged**